server.go

```diff
@@ -1961,10 +1961,22 @@ func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Re
 
 	// Execute each statement.
 	for i, stmt := range q.Statements {
-		// Set default database and policy on the statement.
-		if err := s.NormalizeStatement(stmt, database); err != nil {
-			results.Results[i] = &Result{Err: err}
-			break
+		// If a default database wasn't passed in by the caller,
+		// try to get it from the statement.
+		defaultDB := database
+		if defaultDB == "" {
+			if s, ok := stmt.(influxql.HasDefaultDB); ok {
```

Comment:
Yeah, at first look I think it could just be easier to simply extend the stmt interface. Not sure, but this feels awkward.